src/pages/Apply/Apply.jsx

```diff
@@ -0,0 +1,152 @@
+import React, { useState, useEffect } from "react";
+import { useSelector, useDispatch } from "react-redux";
+
+import { useForm, FormProvider } from "react-hook-form";
+
+import Button from "../../components/Button";
+
+import fetchCourses from "../../services/coursesRequest";
+
+import { setSelectedCourse } from "../../store/actions/coursesActions";
+import applyRequest from "../../services/applyRequest";
+
+import "./apply.scss";
+import paints from "../../assets/images/paints.png";
+
+const Apply = () => {
+  // const [courseId, setCourseId] = useState("");
+  // const [group, setGroup] = useState("");
+
+  const applyData = useSelector((state) => state.courses);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    if (applyData.courses.length === 0) {
+      dispatch(fetchCourses());
+    }
+  }, [fetchCourses]);
+
+  const courses = applyData.courses.map((course) => {
+    return {
+      value: course,
+      display: course.name,
+    };
+  });
+
+  const { register, handleSubmit, watch, getValues } = useForm();
+  const onSubmit = (data) => {
+    applyRequest(data);
+    console.log(data);
+  };
+
+  const methods = useForm({
+    mode: "onChange",
+  });
+
+  const selectedCourse = watch("course");
+
+  const groups = courses.find((course) => course.value.id === selectedCourse)
+    ?.value.groups;
+
+  return (
+    <div className="container">
+      <div className="apply">
+        <div className="apply__form">
+          <h1 className="apply__form-title">
+            Apply to desired course right away!
+          </h1>
+          <p className="apply__form-text">
+            Select desired course and group, leave us your contact information
+            and we’ll reach out to you for confirmation and letting you know
+            about further steps
+          </p>
+          <FormProvider {...methods}>
+            <form onSubmit={handleSubmit(onSubmit)}>
+              <div className="apply__form-first">
+                <div className="apply__form-content">
+                  <h5 className="apply__form-content-text">Course</h5>
+                  <select
+                    className="apply__form-content-input"
+                    {...register("course", { required: true })}
+                  >
+                    <option>Select...</option>
+                    {courses.map((course) => (
+                      <option key={course.value.id} value={course.value.id}>
+                        {course.display}
+                      </option>
+                    ))}
+                  </select>
+                </div>
+                <div className="apply__form-content">
+                  <h5 className="apply__form-content-text">Group</h5>
+                  <select
+                    className="apply__form-content-input"
+                    // disabled={!course}
+                    // value={group}
```

Comment:
remove or uncomment it

Reply:
and i can click on group select if didn't select a couse. But group select should be disabled until we select some course